src/core/core.controller.js

```diff
@@ -43,17 +43,26 @@ module.exports = function(Chart) {
 	function updateConfig(chart) {
 		var newOptions = chart.options;
 
-		// Update Scale(s) with options
-		if (newOptions.scale) {
-			chart.scale.options = newOptions.scale;
-		} else if (newOptions.scales) {
-			newOptions.scales.xAxes.concat(newOptions.scales.yAxes).forEach(function(scaleOptions) {
-				chart.scales[scaleOptions.id].options = scaleOptions;
-			});
-		}
+		// Update Scale(s) or Tooltip with options if needed
+		if (newOptions.scale || newOptions.scales || newOptions.tooltips) {
+			newOptions = helpers.configMerge(
+				Chart.defaults.global,
+				Chart.defaults[chart.config.type],
+				newOptions);
+			chart.options = newOptions;
+			chart.ensureScalesHaveIDs();
+
+			if (newOptions.scale) {
+				chart.scale.options = newOptions.scale;
+			} else if (newOptions.scales) {
+				newOptions.scales.xAxes.concat(newOptions.scales.yAxes).forEach(function(scaleOptions) {
```

Comment:
What happens in the case where new axes are added? I think we'd need to create those scale objects somewhere in here. Similarly, axes could be deleted.